services/web/server/src/simcore_service_webserver/studies_access.py

```diff
@@ -55,41 +55,57 @@ async def get_public_project(app: web.Application, project_uuid: str):
     return prj
 
 
-# TODO: from .users import create_temporary_user
 async def create_temporary_user(request: web.Request):
     """
-        TODO: user should have an expiration date and limited persmissions!
+    TODO: user should have an expiration date and limited persmissions!
     """
     from .login.cfg import get_storage
-    from .login.handlers import ACTIVE, GUEST
+    from .login.handlers import ACTIVE, ANONYMOUS
     from .login.utils import get_client_ip, get_random_string
     from .security_api import encrypt_password
 
-    # from .utils import generate_passphrase
-    # from .utils import generate_password
-
     db = get_storage(request.app)
 
     # TODO: avatar is an icon of the hero!
-    # FIXME: # username = generate_passphrase(number_of_words=2).replace(" ", "_").replace("'", "")
     username = get_random_string(min_len=5)
     email = username + "@guest-at-osparc.io"
     password = get_random_string(min_len=12)
 
+    # creates a user that is marked as ANONYMOUS. see update_user_as_guest
     user = await db.create_user(
         {
             "name": username,
             "email": email,
             "password_hash": encrypt_password(password),
             "status": ACTIVE,
-            "role": GUEST,
+            "role": ANONYMOUS,
```

Comment:
Just to understand what is happenning here. so normally it is impossible for an anonymous user to be registered right? and the GC does not check them?

Reply:
Exactly.

The user is "temporary" anonymous until it is turn into GUEST just after everything was setup correctly and before responding with a redirection
I avoid to create a GUEST from the beginning to avoid that the GC might delete it between this point and when we create the entry in redis.